apps/slackbot/bolt_app.py

```diff
@@ -118,6 +120,45 @@ def get_response(question, previous_messages):
         response = task_agent.run(question)
         return response, None
 
+def initialize_task_agent(question, previous_messages):
+    # This function copeis first section of get_response() and returns task_agent object
+    llm = ChatOpenAI(openai_api_key=cfg.OPENAI_API_KEY, request_timeout=120)
+
+    if cfg.PINECONE_API_KEY:
+        # If pinecone API is specified, then use the Pinecone Database
+        memory = ConversationStore.get_vector_retrieval(
+            cfg.PINECONE_NAMESPACE,
+            cfg.OPENAI_API_KEY,
+            index_name=cfg.PINECONE_INDEX,
+            search_type="similarity_score_threshold",
+            search_kwargs={"score_threshold": 0.0},
+        )
+    else:
+        # use the local Chroma database
+        memory = local_memory
+
+    tools = get_tools(memory)
+
+    task_agent = TaskAgent.from_llm_and_tools(
+        ai_name="Sherpa",
+        ai_role="assistant",
+        ai_id=bot["user_id"],
+        memory=memory,
+        tools=tools,
+        previous_messages=previous_messages,
+        llm=llm,
+    )
+
+    return task_agent
+
+def question_reformat(question):
+    # removes unnecessary words from question, like "-verbose" or "@Sherpa"
```

Comment:
Here you can rename the function so the name says what it does. E.g. `question_without_noise_words`